### ISSUE TITLE
Slider data-value on init, only sets background, not the handle

### ISSUE THREAD

**Issue by openbuilds-engineer** [olton/Metro-UI-CSS#1119]
I have setup a slider like below (from examples, modified a bit)

```
  <input id="fro" class="m-4" data-role="slider" data-hint-mask="Feed: $1%" data-hint-always="true" data-value="50" data-min="0" data-max="100">
```

Once page loads, I get this:
![slider1](https://user-images.githubusercontent.com/31449271/41682027-cbdeec60-74d6-11e8-80cb-68a466fd50dc.JPG)

(Note handle/button still at far left)

I am expecting it to be like this on load?
![expected](https://user-images.githubusercontent.com/31449271/41682041-d9aca2c4-74d6-11e8-933f-65a5ec3983d8.JPG)

I'll be happy to bet I am doing somethiong stupid (: 

Weird observance 1:  If I Resize the window before touching the slider, then button jumps to correct spot
Weird observance 2:  Moving the slider with mouse, or setting 
```
  $('#fro').attr('data-value', 50);
```
after page is loaded, works.

**Comment by olton** [comment]
In Your example min value equals to value `50`

**Comment by olton** [comment]
https://codepen.io/olton/pen/rKdWGb

**Comment by olton** [comment]
In your example min value equals to value
![2018-06-20_23h44_04](https://user-images.githubusercontent.com/365108/41683645-e142f90e-74e3-11e8-9499-77a9988d312f.png)

**Comment by openbuilds-engineer** [comment]
(: Ahh man, since you check these from GMAIL, you didnt get the Edits I applied, pasted the wrong line initially, edited it afterwards. Github.com online view shows the edit.

Regardless, if I take your working Fiddle, and copy paste that into mine, still fails.  
Back to work now, will dig in and see if I can bisect what fails it then. Thanks for the working fiddle, will probably help me isolate what then breaks it, but at least proves its not a bug

**Comment by openbuilds-engineer** [comment]
![crop](https://user-images.githubusercontent.com/31449271/41717314-a26bce24-7559-11e8-8dfe-5b1a161e340c.png)
See, edited (:

**Comment by openbuilds-engineer** [comment]
Okay I have the issue replicated!  

https://codepen.io/anon/pen/WyzLqj shows what happens in my app:

1) The broken Slider is a child inside the DIV
2) In the app, this DIV is only Activated later  (with a JQuery .show() )
3) Once the parent is shown, that's where the problem shows itself

![slider2](https://user-images.githubusercontent.com/31449271/41717801-3da9d542-755b-11e8-9617-ece8a67d4e6f.JPG)

**Comment by openbuilds-engineer** [comment]
Good morning, just checking in if you've had a change to look at the above (: 
I have a deadline today (:

**Comment by olton** [comment]
This problem is due to the fact that the slider can not calculate its position when the `display: none` (for a slider or for a slider parent). I'll try to fix it in the next version. While trying to fix using the visibility property or reset value after displaying parent. Thanks

**Comment by openbuilds-engineer** [comment]
Awesome! Thanks! Will push forward without slider for now (couple buttons
or a menu instead, for now)

**Comment by olton** [comment]
I fix it in 4.2.9 wait for the release on Sunday

**Comment by openbuilds-engineer** [comment]
Wowzers!  That is amazingly fast! Thank you SOOOOOO much!
Loving MetroUI, found it on Tuesday and I am never going back to bootstrap
(;
I really love the way your documentation is structured too - starting off
basic and the  as you work down the guide, it builds upon that.
Not to mention, the support you've given me over the two issues I had - out
of this world! Thank you, from the bottom of my heart!
